zilliqa/src/consensus.rs

```diff
@@ -1694,9 +1742,22 @@ impl Consensus {
             .map(|(qc, _)| qc)
     }
 
-    fn verify_qc_signature(&self, _: &QuorumCertificate) -> Result<()> {
-        // TODO: Build aggregate signature from public keys and validate `qc.block_hash` against `qc.signature`.
-        Ok(())
+    fn verify_qc_signature(
+        &self,
+        qc: &QuorumCertificate,
+        public_keys: Vec<NodePublicKey>,
+    ) -> Result<()> {
+        let len = public_keys.len();
+        match qc.verify(public_keys) {
```

Comment:
the bitvec wasn't quite right when enabling the QC on the first iteration but is now